# Obesity levels 

### Our Motivation 
The "Obesity Levels" dataset has captured our interest because obesity is a major public health issue with significant consequences. By examining this dataset, we hope to uncover the key factors contributing to obesity and leverage machine learning models to enhance our ability to classify and predict obesity levels. Understanding these factors will empower individuals and healthcare professionals to create more effective prevention strategies, ultimately reducing obesity rates and improving public health outcomes. 

### Our Goal 
Our goal is to create a machine-learning model that predicts individual obesity levels using both classification and clustering techniques. The classification approach will enable us to train the model to differentiate between individuals with different obesity levels based on a variety of health, dietary, and lifestyle factors. Clustering, on the other hand, will help identify hidden patterns in the dataset by grouping individuals with similar obesity risk profiles. These clusters will then assist in predicting obesity levels for new individuals by analyzing their resemblance to existing groups.

##### source of dataset : https://www.kaggle.com/datasets/fatemehmehrparvar/obesity-levels

In [1]:
import pandas as pd
df = pd.read_csv('Dataset/ObesityDataSet_raw_and_data_sinthetic.csv')

In [2]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2111 entries, 0 to 2110
Data columns (total 17 columns):
 #   Column                          Non-Null Count  Dtype  
---  ------                          --------------  -----  
 0   Age                             2111 non-null   float64
 1   Gender                          2111 non-null   object 
 2   Height                          2111 non-null   float64
 3   Weight                          2111 non-null   float64
 4   CALC                            2111 non-null   object 
 5   FAVC                            2111 non-null   object 
 6   FCVC                            2111 non-null   float64
 7   NCP                             2111 non-null   float64
 8   SCC                             2111 non-null   object 
 9   SMOKE                           2111 non-null   object 
 10  CH2O                            2111 non-null   float64
 11  family_history_with_overweight  2111 non-null   object 
 12  FAF                             21

#### our dataset includes 17 attributes and 2111 records 

#### 8 attributes of type float and the other 9 of type object 

#### class label : NObeyesdad ( type of obesity ) 

##### Clarifying shortcuts for some attributes :

###### FAVC : Feature, Binary, " Do you eat high caloric food frequently? "

###### FCVC : Feature, Integer, " Do you usually eat vegetables in your meals? "

###### NCP : Feature, Continuous, " How many main meals do you have daily? "

###### CAEC : Feature, Categorical, " Do you eat any food between meals? "

###### CH2O: Feature, Continuous, " How much water do you drink daily? "

###### SCC: Feature, Binary, " Do you monitor the calories you eat daily? "


###### FAF: Feature, Continuous, " How often do you have physical activity? "

###### TUE : Feature, Integer, " How much time do you use technological devices such as cell phone, videogames, television, computer and others? "

###### CALC : Feature, Categorical, " How often do you drink alcohol? "

###### MTRANS : Feature, Categorical, " Which transportation do you usually use? "

###### NObeyesdad : Target, Categorical, "Obesity level"

In [3]:
df.describe()

Age       Height       Weight         FCVC          NCP  \
count  2111.000000  2111.000000  2111.000000  2111.000000  2111.000000   
mean     24.312600     1.701677    86.586058     2.419043     2.685628   
std       6.345968     0.093305    26.191172     0.533927     0.778039   
min      14.000000     1.450000    39.000000     1.000000     1.000000   
25%      19.947192     1.630000    65.473343     2.000000     2.658738   
50%      22.777890     1.700499    83.000000     2.385502     3.000000   
75%      26.000000     1.768464   107.430682     3.000000     3.000000   
max      61.000000     1.980000   173.000000     3.000000     4.000000   

              CH2O          FAF          TUE  
count  2111.000000  2111.000000  2111.000000  
mean      2.008011     1.010298     0.657866  
std       0.612953     0.850592     0.608927  
min       1.000000     0.000000     0.000000  
25%       1.584812     0.124505     0.000000  
50%       2.000000     1.000000     0.625350  
75%       2.477420     1.666678     1.000000  
max       3.000000     3.000000     2.000000

###### This shows a summary of some statistics informations that will help us to know our sample more such as the average age and other attributes , but some are not helpful such as " CH2O " 